app/models/report_link.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class ReportLink < ApplicationRecord
+  belongs_to :from_report, class_name: 'Report', inverse_of: :report_links
+  belongs_to :to_report, class_name: 'Report', inverse_of: :my_report_links
+
+  validates :from_report, uniqueness: { scope: :to_report, message: 'その言及関係の組み合わせは既に存在します。' }
```

Comment:
```suggestion
  validates :from_report, uniqueness: { scope: :to_report }
```

翻訳を変えたいのならi18nの仕組みを活用したいですね。

https://railsguides.jp/i18n.html#%E3%82%A8%E3%83%A9%E3%83%BC%E3%83%A1%E3%83%83%E3%82%BB%E3%83%BC%E3%82%B8%E3%81%AE%E3%82%B9%E3%82%B3%E3%83%BC%E3%83%97